test/functional/tests/fault_injection/test_fault_injection_opencas_load.py

```diff
@@ -0,0 +1,139 @@
+        stop_n_load_cache_fs(cache_mode, filesystem, cache_dev, core_dev)
+        stop_n_load_cache_notfs(cache_mode, cache_dev, core_dev)
+
+
+def stop_n_load_cache_fs(cache_mode, filesystem, cache_dev, core_dev):
+    with TestRun.step("Start cache."):
+        cache = casadm.start_cache(cache_dev, cache_mode, force=True)
+
+    with TestRun.step("Add core to cache with xfs filesystem and mount it."):
+        core_dev.create_filesystem(filesystem)
+        core = cache.add_core(core_dev)
+        core.mount(mount_point)
+
+    with TestRun.step(f"Create test file in /tmp directory."):
+        file = fs_utils.create_test_file('/tmp/test_file',
+                                         'There is nothing interesting inside this file.')
+
+    with TestRun.step("Copy file to cache device"):
+        copied_file = File.copy(file.full_path, test_file_path, force=True)
+
+    with TestRun.step("Unmount cache's exported object."):
+        core.unmount()
+
+    with TestRun.step("Stop cache with option '-n'."):
+        cache.stop(no_data_flush=True)
+        caches_count = len(casadm_parser.get_caches())
+        if caches_count != 0:
+            TestRun.fail(f"Expected caches count: 0; Actual caches count: {caches_count}.")
+
+    with TestRun.step("Try start cache without loading metadata."):
+        output = TestRun.executor.run_expect_fail(cli.start_cmd(cache_dev=str(cache_dev),
+                                                                cache_mode=str(cache_mode),
+                                                                force=False, load=False))
+        if not output.stderr:
+            TestRun.fail("Cache start succeeded (should fail)!")
+
+    with TestRun.step("Load cache."):
+        cache = casadm.load_cache(cache.cache_device)
+        caches_count = len(casadm_parser.get_caches())
+        if caches_count != 1:
+            TestRun.fail(f"Expected caches count: 1 Actual caches count: {caches_count}.")
+        cores_count = len(casadm_parser.get_cores(cache.cache_id))
+        if cores_count != 1:
+            TestRun.fail(f"Expected cores count: 1; Actual cores count: {cores_count}.")
+
+    with TestRun.step("Mount cache's exported object."):
+        core.mount(mount_point)
+
+    with TestRun.step("Compare properties of test file before and after copying."):
+        if file.get_properties() != copied_file.get_properties():
+            TestRun.LOGGER.error("File properties before copying and after are different.")
+
+    with TestRun.step("Unmount cache's exported object."):
+        core.unmount()
+
+    with TestRun.step("Stop all caches."):
+        casadm.stop_all_caches()
+
+
+def stop_n_load_cache_notfs(cache_mode, cache_dev, core_dev):
+    with TestRun.step("Start cache."):
+        cache = casadm.start_cache(cache_dev, cache_mode, force=True)
+
+    with TestRun.step("Add core device without filesystem."):
+        cache.add_core(core_dev)
+
+    with TestRun.step("Copy data to cache device"):
```

Comment:
What for data is copied here?

Reply:
This is step from original test. I think it's to fill cache with data, which would not be flushed.
Also thanks, because I saw I done operation on wrong device.